gst/nnstreamer/tensor_query/tensor_query_serversrc.c

```diff
@@ -290,62 +264,143 @@ gst_tensor_query_serversrc_get_property (GObject * object, guint prop_id,
-    return FALSE;
-  }
+  src->edge_h = gst_tensor_query_server_get_edge_handle (src->server_h);
+  nns_edge_set_info (src->edge_h, "IP", src->host);
+  port = g_strdup_printf ("%d", src->port);
+  nns_edge_set_info (src->edge_h, "PORT", port);
+  g_free (port);
 
   /** Publish query sever connection info */
-  if (src->operation) {
-    tensor_query_hybrid_set_node (&src->hybrid_info, src->host, src->port,
-        src->server_info_h);
-    tensor_query_hybrid_set_broker (&src->hybrid_info,
-        src->broker_host, src->broker_port);
+  if (src->topic) {
+    nns_edge_set_info (src->edge_h, "TOPIC", src->topic);
+    tensor_query_hybrid_set_node (&src->hybrid_info, src->srv_host,
+        src->srv_port, src->server_info_h);
+    tensor_query_hybrid_set_broker (&src->hybrid_info, src->host, src->port);
 
-    if (!tensor_query_hybrid_publish (&src->hybrid_info, src->operation)) {
+    if (!tensor_query_hybrid_publish (&src->hybrid_info, src->topic)) {
       nns_loge ("Failed to publish a topic.");
       return FALSE;
     }
   } else {
+    g_free (src->srv_host);
+    src->srv_host = g_strdup (src->host);
+    src->srv_port = src->port;
     nns_logi ("Query-hybrid feature is disabled.");
     nns_logi
-        ("Specify operation to register server to broker (e.g., operation=object_detection/mobilev3).");
+        ("Specify topic to register server to broker (e.g., topic=object_detection/mobilev3).");
+  }
+
+  if (0 != nns_edge_start (src->edge_h, true)) {
+    nns_loge
+        ("Failed to start NNStreamer-edge. Please check server IP and port");
+    return FALSE;
+  }
+  nns_edge_set_event_callback (src->edge_h, _nns_edge_event_cb, src);
+  if (!gst_tensor_query_server_wait_sink (src->server_h)) {
+    nns_loge ("Failed to get server information from query server.");
+    return FALSE;
   }
+
   return TRUE;
 }
 
 /**
- * @brief stop processing of query_serversrc
+ * @brief Get buffer from message queue.
  */
-static gboolean
-gst_tensor_query_serversrc_stop (GstBaseSrc * bsrc)
+static GstBuffer *
+_gst_tensor_query_serversrc_get_buffer (GstTensorQueryServerSrc * src)
 {
-  GstTensorQueryServerSrc *src = GST_TENSOR_QUERY_SERVERSRC (bsrc);
+  nns_edge_data_h data_h = NULL;
+  GstBuffer *buffer = NULL;
+  guint i, num_data;
+  GstMetaQuery *meta_query;
 
-  tensor_query_hybrid_close (&src->hybrid_info);
-  nnstreamer_query_server_data_free (src->server_data);
-  src->server_data = NULL;
-  return TRUE;
+  data_h = g_async_queue_pop (src->msg_queue);
+
+  if (!data_h) {
+    nns_loge ("Failed to get message from the server message queue");
+    return NULL;
+  }
+  buffer = gst_buffer_new ();
+
+  if (NNS_EDGE_ERROR_NONE != nns_edge_data_get_count (data_h, &num_data)) {
+    nns_loge ("Failed to get the number of memories of the edge data.");
+    gst_buffer_unref (buffer);
+    return NULL;
+  }
+  for (i = 0; i < num_data; i++) {
+    void *data = NULL;
+    size_t data_len = 0;
+
+    nns_edge_data_get (data_h, i, &data, &data_len);
```

Comment:
crash? 
1. nns_edge_data_get() returns memory ptr in data handle, need memdup() to append this into gst-buffer.
2. need destroy edge data handle after getting edge data from queue.